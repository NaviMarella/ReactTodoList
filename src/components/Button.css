.btn {
    border: none;
    color: white;
    overflow: hidden;
    padding: 0;
    text-transform: uppercase;
    width: 150px;
    height: 50px;
    margin: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
}

.btn-success {
    background-color: green;
}

.btn-large{
    font-size: 20px;
}
