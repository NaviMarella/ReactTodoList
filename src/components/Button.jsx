import React from "react";
import "./Button.css";

const Button = ({type, text, children, handleButtonClick}) => {

    function handleClick(e) {
       e.preventDefault();
       handleButtonClick();
    }

    return (
        <button type={type} className='btn btn-large btn-success' onClick={handleClick}>
            {children}
            {text}
        </button>
    )
}

export default Button;