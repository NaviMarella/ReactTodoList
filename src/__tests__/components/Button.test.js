import { shallow } from 'enzyme';
import React from 'react';
import Button from "../../components/Button";

describe('Button Component', () => {
  test('render snapshot', () => {
      let props = {
        type: "submit",
        text: "Test Button",
        handleButtonClick: jest.fn()
      }
    const wrapper = shallow(<Button {...props} />);
     wrapper.simulate('click', { preventDefault() {} });
     expect(props.handleButtonClick).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();

  });
});
