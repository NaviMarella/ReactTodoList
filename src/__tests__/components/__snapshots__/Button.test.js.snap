// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    handleButtonClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    text="Test Button"
    type="submit"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        "Test Button",
      ],
      "className": "btn btn-large btn-success",
      "onClick": [Function],
      "type": "submit",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      "Test Button",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          "Test Button",
        ],
        "className": "btn btn-large btn-success",
        "onClick": [Function],
        "type": "submit",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        "Test Button",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
